Add guide for creating a new app (#86)
ONSdigital/dp: guides/NEW_APP.md

Creating a new app
==================

> The following steps will guide you through add adding a new app to our system that is built by the CI system and deployed to the live environment.

Reviewer: ```suggestion
The following steps will guide you through adding a new app to our system that is built by the CI system and deployed to the live environment.
```

Prerequisites
-------------

> In order to follow this guide you will need:

Reviewer: Really really pedantic and can be ignored

```suggestion
To follow this guide you will need:
```
- Keeping the wording short and to the point, wherever possible makes it quicker and easier to read.

> * A github account with access to the ONSdigital organisation

Reviewer: ```suggestion
* A Github account with access to the ONSdigital organisation
```
- Name of the tool; capital letter

* A login for the concourse CI environment

* [dp-cli](https://github.com/ONSdigital/dp-cli)

* Git:

  ```
  brew install git
  ```

* Terraform >= 0.12:

  ```
  brew install terraform
  ```

> * Fly cli (can be downloaded from the links in the corner of the concourse UI)

Reviewer: ```suggestion
* Fly CLI (can be downloaded from the links in the corner of the concourse UI)
```
Initialism in capitals (concourse refers to it in caps too, https://concourse-ci.org/fly.html )

Getting started
---------------

> These steps will create a new app that is built by CI and deployed to the develop and production environments.  If any of the following github links 404 it is likely that you need to be added to the ONSdigital organisation.

Reviewer: ```suggestion
These steps will create a new app that is built by CI and deployed to the Develop and Production environments.  If any of the following Github links 404 likely you need to be added to the ONSdigital organisation.
```
- 'Github' is a name, capital letter
- Our environments are named 'Production' and 'Develop', capital letter
- "it is likely that you need" -> 'likely you need'; makes the sentence flow easier

Author: Our environments are usually referred to as `develop` and `production` (lower case) so I have left the casing as is.

> 1. Determine what language you want to use and technology your app will likely use

Reviewer: - Bit pedantic again the sentence could be optimised a bit maybe something like
```suggestion
1. Determine what language and technologies you will use
```
- The general theme of this guide is to be informative and this list is a step-by-step set of instructions. So maybe worth avoiding words like 'want' and 'likely'.

2. Pick a local dev port for your app and add it to the [local ports list](PORTS.md)

3. Use the [`dp-cli`](https://github.com/ONSdigital/dp-cli) to generate the new repo and populate it with the standard boilerplate files:

   ```

>    dp-utils generate-project

Reviewer: So yes this will generate a project but won't add it to Github necessarily, there is a flag that needs to be provided. I would suggest linking to the readme.md here but that is still in a PR - this will be where it lives mind, but I can come back and PR and edit this later.
https://github.com/ONSdigital/dp-cli/blob/fbc8fa72b1c4052541c2943efde2af56655c3237/project-generation/COMPLETE_PROJECT_SETUP.md

Author: Adding a link to where that README will be. Does mean it is broken in the short term just until your PR is merged.

   ```

4. [Add the new app to the environment](https://github.com/ONSdigital/dp-setup#adding-a-new-app)

5. [Add the new app to CI](https://github.com/ONSdigital/dp-ci#adding-a-new-app-to-ci)

> 6. Check that your new app has successfully deployed to develop

Reviewer: Considering the level of detail this guide goes to, it might be worth adding how to check that the app has been deployed (I imagine check concourse) rather than just stating check it.
